File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int n = nums.size();
        vector<int> T(n, 0);
        T[0] = nums[0];
        int maxsum = T[0];
        for (int i = 1; i < n; ++i) {
            T[i] = max(T[i-1] + nums[i], nums[i]);  
            maxsum = max(T[i], maxsum);
        }
        return maxsum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_sum = 0
        largest_sum = nums[0]
        for num in nums:
            curr_sum += num
            largest_sum = max(largest_sum, curr_sum)
            if curr_sum < 0:
                curr_sum = 0
        return largest_sum